### ISSUE TITLE
Collection does not work properly with associative arrays

### ISSUE THREAD

**Issue by ramonacat** [phpextra/common#2]
If collection is constructed with associative array as an argument (e.g. new Collection(array('a' => 1, 'b' => 2))) it is impossible to iterate it via foreach (it works as if it were an empty collection).

If this case is not supported, we should either call array_values on input array or throw an exception if it is associative.

**Comment by jkobus** [comment]
True. Foreach or array values should work in this case. Waiting for a PR.
